packages/rest/src/__tests__/acceptance/validation/validation.acceptance.ts

```diff
@@ -84,6 +99,21 @@ describe('Validation at REST level', () => {
         .expect(400);
     });
 
+    it('rejects requests with empty required object', async () => {
+      const DATA = {
+        name: 'iPhone',
+        description: null,
+        price: 10,
+        category: null,
+      };
+
+      await client
+        .post('/products')
+        .type('json')
+        .send(DATA)
+        .expect(422);
+    });
```

Comment:
I am afraid this test is passing on the current master too. Here is the error reported:

```js
{
  statusCode: 422,
  name: 'UnprocessableEntityError',
  message: 'The request body is invalid. See error object `details` property for more info.',
  code: 'VALIDATION_FAILED',
  details: [
    {
      path: '',
      code: 'additionalProperties',
      message: 'should NOT have additional properties',
      info: [Object]
    }
  ]
}
```

@noels PTAL. 

We need a test that fails on the current master and thus reproduces the bug you are trying to fix. Without such test, we have very little confidence that your changes are actually achieving the desired goal, plus it's easy to introduce unwanted regressions in the future.

Reply:
HI @bajtos Thanks for having a look.  I reset my local copy to the upstream master, replaced the validation.acceptance.ts file with the one from my branch, then ran mocha in loopback-next/packages/rest
and got:
Error: expected 422 "Unprocessable Entity", got 200 "OK"
    at Test._assertStatus (/home/noels/Documents/workspace/ho/GEFEAPI/loopback-next/packages/testlab/node_modules/supertest/lib/test.js:268:12)
    at Test._assertFunction (/home/noels/Documents/workspace/ho/GEFEAPI/loopback-next/packages/testlab/node_modules/supertest/lib/test.js:283:11)
    at Test.assert (/home/noels/Documents/workspace/ho/GEFEAPI/loopback-next/packages/testlab/node_modules/supertest/lib/test.js:173:18)
    at localAssert (/home/noels/Documents/workspace/ho/GEFEAPI/loopback-next/packages/testlab/node_modules/supertest/lib/test.js:131:12)
    at /home/noels/Documents/workspace/ho/GEFEAPI/loopback-next/packages/testlab/node_modules/supertest/lib/test.js:128:5
    at Test.Request.callback (/home/noels/Documents/workspace/ho/GEFEAPI/loopback-next/packages/testlab/node_modules/superagent/lib/node/index.js:716:12)
    at parser (/home/noels/Documents/workspace/ho/GEFEAPI/loopback-next/packages/testlab/node_modules/superagent/lib/node/index.js:916:18)
    at IncomingMessage.res.on (/home/noels/Documents/workspace/ho/GEFEAPI/loopback-next/packages/testlab/node_modules/superagent/lib/node/parsers/json.js:19:7)
    at endReadableNT (_stream_readable.js:1059:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
Did you npm run build before running the test? Am I missing something?

Reply:
That's weird, let me try again.

Reply:
I created a feature branch based on the current master (as of today morning), adding your commit with the test (b44cac5).

You can find the feature branch here:

https://github.com/strongloop/loopback-next/tree/test-pr-3828

When I run `npm it` (install & test), all tests are green.

I think the change in behavior may be caused by https://github.com/strongloop/loopback-next/pull/3895, where we started to add `additionalProperties: false` to schema emitted for models in strict mode (which is the default).

When I change the definition of `Product` model in the acceptance test and disable strict mode, then your new test fails as expected. (Unfortunately, there is one more test that fails too.)

```patch
diff --git a/packages/rest/src/__tests__/acceptance/validation/validation.acceptance.ts b/packages/rest/src/__tests__/acceptance/validation/validation.acceptance.ts
index 19697b83f..9070273fa 100644
--- a/packages/rest/src/__tests__/acceptance/validation/validation.acceptance.ts
+++ b/packages/rest/src/__tests__/acceptance/validation/validation.acceptance.ts
@@ -28,7 +28,7 @@ describe('Validation at REST level', () => {
   let app: RestApplication;
   let client: Client;

-  @model()
+  @model({settings: {strict: false}})
   class Product {
     @property()
     id: number;
```

Obviously, we don't want to change all existing tests to start testing the scenario with `strict: false` and no longer test the default setup with `strict: true`. To support non-strict mode, we need to create a new group of tests in `validation.acceptance.ts` that will use a new model with `strict: false`.

Let me ask you a question: how are your models configured? Do you use the default "strict" model behavior, or are you disabling it via model settings? If you are using strict mode, then I think you should be find a this pull request is not needed for your app.